src/dimension.c

```diff
@@ -879,6 +897,23 @@ ts_dimension_set_chunk_interval(Dimension *dim, int64 chunk_interval)
 	return dimension_scan_update(dim->fd.id, dimension_tuple_update, dim, RowExclusiveLock);
 }
 
+static inline int64
+interval_to_usec(Interval *interval)
+{
+	return (interval->month * DAYS_PER_MONTH * USECS_PER_DAY) + (interval->day * USECS_PER_DAY) +
+		   interval->time;
+}
+
+int
+ts_dimension_set_compress_interval(Dimension *dim, int64 compress_interval)
+{
+	Assert(IS_OPEN_DIMENSION(dim));
```

Comment:
Since this is reading from the metadata, it might make sense to generate an error if the dimension is not open. Otherwise, you might proceed with this code for release builds, causing hard-to-find bugs.